src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ClientCreds/JsonWebToken.cs

```diff
@@ -217,10 +218,7 @@ public string X509CertificateThumbprint
                     return this.Credential.Thumbprint;
                 }
 
-                set
-                {
-                    // This setter is required by DataContractJsonSerializer
-                }
+                set { value = this.Credential.Thumbprint; }
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable 'value'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS1854)